File: sample/src/main/java/com/davidmiguel/sample/Card1Activity.java
package com.davidmiguel.sample;

import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;

import com.davidmiguel.dragtoclose.DragListener;
import com.davidmiguel.dragtoclose.DragToClose;

import androidx.appcompat.app.AppCompatActivity;

/**
 * In this example the activity is closed when the card is dragged out.
 * Dragging events are logged.
 */
public class Card1Activity extends AppCompatActivity {

    private static final String TAG = Card1Activity.class.getSimpleName();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_card1);

        final DragToClose dragToClose = findViewById(R.id.drag_to_close);
        dragToClose.setDragListener(new DragListener() {
            @Override
            public void onStartDraggingView() {
                Log.d(TAG, "onStartDraggingView()");
            }

            @Override
            public void onViewCosed() {
                Log.d(TAG, "onViewCosed()");
            }

            @Override
            public void onDragging(float dragOffset) {
                Log.d(TAG, String.format("onDragging(%.2f)", dragOffset));
            }
        });
        Button btn = findViewById(R.id.btn);
        btn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                dragToClose.closeDraggableContainer();
            }
        });
    }
}

File: sample/src/main/java/com/davidmiguel/sample/Card1Activity.kt
package com.davidmiguel.sample

import android.os.Bundle
import android.util.Log
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity
import com.davidmiguel.dragtoclose.DragListener
import com.davidmiguel.dragtoclose.DragToClose

/**
 * In this example the activity is closed when the card is dragged out.
 * Dragging events are logged.
 */
class Card1Activity : AppCompatActivity() {

    private val tag = "Card1Activity"

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_card1)

        val dragToClose = findViewById<DragToClose>(R.id.drag_to_close)
        dragToClose.setDragListener(object : DragListener {
            override fun onStartDraggingView() {
                Log.d(tag, "onStartDraggingView()")
            }

            override fun onDragging(dragOffset: Float) {
                Log.d(tag, "onDragging(): $dragOffset")
            }

            override fun onViewCosed() {
                Log.d(tag, "onViewCosed()")
            }
        })
        val btn = findViewById<Button>(R.id.btn)
        btn.setOnClickListener { dragToClose.closeDraggableContainer() }
    }
}
